hack/verify-go-src.sh

```diff
@@ -99,10 +48,22 @@ function run-checks {
   done
 }
 
-if ${SILENT} ; then
-  echo "Running in silent mode, run with -v if you want to see script logs."
+# Set up the temporary GOPATH
+mkdir -p "${TMP_REPO_ROOT}"
+cp -r "${REPO_ROOT}/"* "${TMP_REPO_ROOT}"
+export GOPATH="${TMP_GOPATH}"
```

Comment:
Why the temporary `GOPATH`?

Reply:
This helps to run this script in a vanilla Prow environment because some of the scripts called by this one e.g. `verify-govet.sh`, run `go list` which will print the import path for each package (e.g. `k8s.io/cluster-registry/...`) to pass into `go vet` and `go` will not find it correctly in `GOPATH` unless the package exists using that same import path. So this temporary `GOPATH` is set up to replicate the same import path location.

I will add a comment to this effect.